lib/Conversion/TorchToLinalg/TorchToLinalg.cpp

```diff
@@ -2508,6 +2532,107 @@ class ConvertAtenPermuteOp : public OpConversionPattern<AtenPermuteOp> {
+  ConvertAtenSliceLikeOp(TypeConverter &typeConverter, MLIRContext *context)
+      : ConversionPattern(typeConverter, MatchAnyOpTypeTag(), /*benefit=*/1,
+                          context) {}
+  LogicalResult
+  matchAndRewrite(Operation *op, ArrayRef<Value> operands,
+                  ConversionPatternRewriter &rewriter) const override {
+    if (!isa<AtenSliceTensorOp, AtenSelectIntOp>(op))
+      return rewriter.notifyMatchFailure(op, "not a supported slice op");
+    if (failed(verifyLinalgCompatibleTypes(op, rewriter)))
+      return failure();
+
+    Location loc = op->getLoc();
+    TypeConverter *typeConverter = getTypeConverter();
+
+    auto input = operands[0];
+    RankedTensorType inputType = input.getType().cast<RankedTensorType>();
+    RankedTensorType resultType =
+        typeConverter->convertType(op->getResult(0).getType())
+            .cast<RankedTensorType>();
+    int64_t rank = resultType.getRank();
+    SmallVector<Value> resultShape;
+    Value zero = rewriter.create<arith::ConstantIndexOp>(loc, 0);
+    Value one = rewriter.create<arith::ConstantIndexOp>(loc, 1);
+
+    SmallVector<Value> offsets, strides;
+    strides.resize(inputType.getRank(), one);
+    for (int64_t i = 0; i < rank; i++)
+      strides[i] = one;
+    offsets.resize(inputType.getRank(), zero);
+    if (auto sliceOp = dyn_cast<AtenSliceTensorOp>(op)) {
+      AtenSliceTensorOp::Adaptor adaptor(operands);
+      Value end, start, stepIndex;
+      resultShape = getTensorSizes(rewriter, loc, input);
+      int64_t dim, step;
+      if (!matchPattern(sliceOp.dim(), m_TorchConstantInt(&dim)))
+        return op->emitError("unimplemented: dim is not constant");
+      if (adaptor.start().getType().isa<Torch::NoneType>()) {
+        start = zero;
+      } else {
+        auto temp = castIndexToInt(rewriter, loc, resultShape[dim]);
+        start = toPositiveDimDynamic(rewriter, loc, adaptor.start(), temp);
+        Value cst0 = rewriter.create<arith::ConstantOp>(
+            loc, rewriter.getZeroAttr(temp.getType()));
+        Value predDimGEZero = rewriter.create<arith::CmpIOp>(
+            loc, arith::CmpIPredicate::sge, start, cst0);
+        start = rewriter.create<SelectOp>(loc, predDimGEZero, start, cst0);
+        start = castIntToIndex(rewriter, loc, start);
+      }
+      if (adaptor.end().getType().isa<Torch::NoneType>()) {
+        end = resultShape[dim];
+      } else {
+        auto temp = castIndexToInt(rewriter, loc, resultShape[dim]);
+        end = toPositiveDimDynamic(rewriter, loc, adaptor.end(), temp);
+        end = castIntToIndex(rewriter, loc, end);
+      }
+      if (!matchPattern(sliceOp.step(), m_TorchConstantInt(&step))) {
+        if (adaptor.step().getType().isa<Torch::NoneType>()) {
+          step = 1;
+        } else {
+          return op->emitError("unimplemented: step is not constant");
+        }
+      }
+      // Slice logic: resultSize = floordiv(end - start + step - 1,  step)
```

Comment:
it could also be possible that end given by the user is smaller than start. we could add a runtime assert saying we only support end <= start for now. I think if there is complication with `end == start` we could drop support for `end == start` for this version as well.

Reply:
The issue seems to stem from the fact that the output results in a 0-dimensional view and that breaks in the refbackend. I think the lowering itself ends up being correct

Reply:
**a =torch.randint(10, (5,5))
>>> print(a)
tensor([[5, 7, 2, 6, 5],
        [3, 5, 2, 1, 2],
        [6, 9, 6, 7, 5],
        [8, 3, 9, 6, 2],
        [7, 9, 9, 3, 7]])
>>> print(a[:0,:0])
tensor([], size=(0, 0), dtype=torch.int64)**

Reply:
For now I'm going to leave it as xfail and open an issue about whether or not to deal with empty tensors in the refbackend

Reply:
Sounds good! Thanks for opening the issue. Can you refer to the issue https://github.com/llvm/torch-mlir/issues/448  in the comment of xfail?

Reply:
I added it as a comment on the test, do you mean on the actual xfail_sets.py?